File: miniRuby/interpreter/command/AssignCommand.cpp
#include <iostream>
#include <list>

#include "AssignCommand.h"
#include "../expr/Expr.h"

AssignCommand::AssignCommand(int line, std::list<Expr*> left, std::list<Expr*> right)
	: Command(line), m_left(left), m_right(right){
}

AssignCommand::~AssignCommand() {
    for (std::list<Expr*>::iterator it = m_left.begin(),
	     ed = m_left.end(); it != ed; it++) {
		Expr* cmd = *it;
		delete cmd;
	}
    for (std::list<Expr*>::iterator it = m_right.begin(),
	     ed = m_right.end(); it != ed; it++) {
		Expr* cmd = *it;
		delete cmd;
	}
	// delete m_left;
	// delete m_right;
}

void AssignCommand::execute() {
    /*
    for (std::list<Expr*>::iterator it = m_left.begin(),
	     ed = m_left.end(); it != ed; it++) {
		Expr* cmd = *it;
        m_left->setValue(value);
		//cmd->execute();
	}
    for (std::list<Expr*>::iterator it = m_right.begin(),
	     ed = m_right.end(); it != ed; it++) {
		Expr* cmd = *it;
        int value = m_right->expr();
		//cmd->execute();
	}
	// int value = m_right->expr();
	//m_left->setValue(value);*/
}

File: miniRuby/JAVA/interpreter/command/AssignCommand.java
package interpreter.command;

import java.util.ArrayList;
import java.util.List;

import interpreter.expr.Expr;
import interpreter.expr.SetExpr;
import interpreter.util.Exit;
import interpreter.value.Value;

public class AssignCommand extends Command{
    private List<Expr> left;
    private List<Expr> right;

    public AssignCommand(int line, List<Expr> left, List<Expr> right){
        super(line);
        this.left = left;
        this.right = right;
    }

    @Override
    public void execute(){
        if(left.size() != right.size()){
            Exit.exit(super.getLine());
        }else {
            List<Value<?>> list = new ArrayList<Value<?>>();

            for(int i = 0; i < left.size(); i++){
                list.add(right.get(i).expr());
            }

            for(int i = 0; i < left.size(); i++){
                if(left.get(i) instanceof SetExpr){
                    SetExpr setExpr = (SetExpr) left.get(i);

                    setExpr.setValue(list.get(i));
                }else {
                    Exit.exit(super.getLine());
                }
            }
        }
    }
}
